java/src/client/join/JoinGameController.java

```diff
@@ -117,9 +117,20 @@ public void reloadGamesList() {
                 .onSuccess(games -> SwingUtilities.invokeLater(() -> {
                     getJoinGameView().setGames(games, getGameManager().getPlayerInfo());
                     if (!getNewGameView().isModalShowing() && !getSelectColorView().isModalShowing()) {
-                        //TODO fix this whack redrawing stuff
                         getJoinGameView().closeModal();
                         getJoinGameView().showModal();
+                    } else if (getSelectColorView().isModalShowing() && selectedGame != null) {
+                        for (GameInfo g : games) {
+                            if (g.getId() == selectedGame.getId()) {
+                                selectedGame = g;
+                                break;
+                            }
+                        }
+                        selectedGame.getPlayers().stream()
+                                .filter(p -> p.getId() != getGameManager().getPlayerInfo().getId())
+                                .forEach(p -> getSelectColorView().setColorEnabled(p.getColor(), false));
```

Comment:
Will this handle when the active colors change? For example, Player 1 joins with blue, then Player 2 joins and sees that blue is not taken, then Player 1 leaves and  rejoins with green, Player 2 should now be able to choose blue and not green. Not super important right now, but maybe a TODO should belong here.